table/tables/partition_test.go

```diff
@@ -700,6 +702,1550 @@ func TestIssue31629(t *testing.T) {
+		fmt.Println(i, ":", ddlSQL)
+		executeSQLWrapper(t, tk, ddlSQL)
+		// insert data
+		for _, insertsql := range insertSQLs {
+			executeSQLWrapper(t, tk, insertsql)
+		}
+		// execute testcases
+		for j, selInfo := range testCase.selectInfo {
+			fmt.Println(j, ":", selInfo.selectSQL)
+			tk.MustQuery(selInfo.selectSQL).Check(testkit.Rows(strconv.Itoa(selInfo.rowCount)))
+			if selInfo.executeExplain {
+				result := tk.MustQuery("EXPLAIN " + selInfo.selectSQL)
+				if selInfo.point {
+					result.CheckContain("Point_Get")
+				}
+				if selInfo.batchPoint {
+					result.CheckContain("Batch_Point_Get")
+				}
+				if selInfo.pruned {
+					for _, part := range selInfo.usedPartition {
+						result.CheckContain(part)
+					}
+					for _, part := range selInfo.notUsedPartition {
+						result.CheckNotContain(part)
+					}
+				}
+			}
+		}
+		executeSQLWrapper(t, tk, dropSQL)
+	}
+}
+
+func executeSQLWrapper(t *testing.T, tk *testkit.TestKit, SQLString string) {
+	res, err := tk.Exec(SQLString)
+	if res != nil {
+		res.Close()
+	}
+	require.Nil(t, err)
+}
+
+func TestKeyPartitionTableBasic(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("create database partitiondb")
+	defer tk.MustExec("drop database partitiondb")
+	tk.MustExec("use partitiondb")
+	testCases := []struct {
+		createSQL  string
+		dropSQL    string
+		insertSQL  string
+		selectInfo []compoundSQL
+	}{
+		{
+			createSQL: "CREATE TABLE tkey0 (col1 INT NOT NULL, col2 DATE NOT NULL, col3 INT NOT NULL, col4 INT NOT NULL,UNIQUE KEY (col3)) PARTITION BY KEY(col3) PARTITIONS 4",
+			selectInfo: []compoundSQL{
+				{
+					"SELECT count(*) FROM tkey0",
+					false, false, false, false, []string{}, []string{}, 4,
+				},
+				{
+					"SELECT count(*) FROM tkey0 PARTITION(p0)",
+					false, false, false, false, []string{}, []string{}, 1,
+				},
+				{
+					"SELECT count(*) FROM tkey0 PARTITION(p1)",
+					false, false, false, false, []string{}, []string{}, 1,
+				},
+				{
+					"SELECT count(*) FROM tkey0 PARTITION(p2)",
+					false, false, false, false, []string{}, []string{}, 0,
+				},
+				{
+					"SELECT count(*) FROM tkey0 PARTITION(p3)",
+					false, false, false, false, []string{}, []string{}, 2,
+				},
+				{
+					"SELECT count(*) FROM tkey0 WHERE col3 = 3",
+					true, false, true, true, []string{"partition:p3"}, []string{"partition:p0", "partition:p1", "partition:p2"}, 1,
+				},
+				{
+					"SELECT count(*) FROM tkey0 WHERE col3 = 3 or col3 = 4",
+					false, false, true, true, []string{"partition:p0", "partition:p3"}, []string{"partition:p1", "partition:p2"}, 2,
+				},
+				{
+					"SELECT count(*) FROM tkey0 WHERE col3 >1 AND col3 < 4",
+					false, false, true, true, []string{"partition:p1", "partition:p3"}, []string{"partition:p0", "partition:p2"}, 2,
+				},
+			},
+			insertSQL: "INSERT INTO tkey0 VALUES(1, SYSDATE(), 1, 1), (2, SYSDATE(), 2, 2), (3, SYSDATE(), 3, 3), (4, SYSDATE(), 4, 4)",
```

Comment:
Please move the insertSQL between the createSQL and selectInfo for this and following test cases, that makes it much easier to read the test cases.